.github/workflows/publish-package.yml

```diff
@@ -36,11 +45,25 @@ jobs:
       - name: Build
         run: yarn build
 
-      # TODO: we still need to make the tests work in CI environments
-      # will come back to this
-      # - name: Test
-      #   run: yarn test
+      - name: E2E tests
+        id: e2e
+        env:
+          API_KEY: ${{ secrets.API_KEY }}
+          OWNER_MNEMONIC: ${{ secrets.OWNER_MNEMONIC }}
+          PAYMASTER_POLICY_ID: ${{ secrets.PAYMASTER_POLICY_ID }}
+        run: yarn test:e2e
+
+      - name: Artifact of E2E tests
+        id: artifact
+        if: steps.e2e.outcome == 'success'
+        run: echo 'e2e=true' >> "$GITHUB_OUTPUT"
 
-      # Lerna publish will, compute the new version, update files, run precommit hooks, tag, publish, and push change log
+  # Lerna publish will, compute the new version, update files, run precommit hooks, tag, publish, and push change log
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: build_and_test
+    if: inputs.publish && (needs.build_and_test.outputs.artifact == 'true')
+    steps:
```

Comment:
does this not need checkout, install, and build? I think it does because the artifacts don't transfer to the next job?

Reply:
i just realized i can just do the conditionals on a job-step, not just jobs. so just going to consolidate deploy into the build_and_test job to become build_and_test_and_deploy, conditional on the publish flag and successful e2e. 

this avoids duplicating checkout, install, and build.